src/Microsoft.TestPlatform.Client/TestPlatform.cs

```diff
@@ -157,7 +157,7 @@ public TestPlatform()
         }
 
         /// <inheritdoc/>
-        public void StartTestSession(
+        public bool StartTestSession(
```

Comment:
Right now there's no enforcing of the parallel level, we just start as many testhosts as required by the `StartTestSession` caller and this call indeed blocks until all of them are started. This may not necessarily be transparent to the wrapper caller if they use the async APIs though.

When we'll introduce the LRU cache we have discussed about, the call will still block until all testhosts up to the maximum number of hosts allowed have been started.

Reply:
Currently there's no timeline for the LRU cache implementation, but I expect it'll land at some point in TP 17.2.

I expect there'll be no changes for the caller the way I'm thinking of the implementation now. The fact that such a LRU cache exists in the first place shouldn't be apparent to the caller, instead he may notice small perf inconsistencies (i.e. even though the caller is using a test session there will be times when we cannot guarantee we can use a pre-started testhost, instead it'll be a best effort kind of strategy).